R/derive_params_growth_age.R

```diff
@@ -0,0 +1,235 @@
+#'  defines the parameter code and parameter.
+#'
+#' *Permitted Values*: List of variable-value pair
+#'
+#'  If left as default value, `NULL`, then parameter not derived in output dataset
+#'
+#' @return The input dataset additional records with the new parameter added.
+#'
+#'
+#' @family der_growth
+#'
+#' @keywords der_growth
+#'
+#' @export
+#'
+#' @examples
+#' library(dplyr)
+#' library(lubridate)
+#' library(rlang)
+#' library(admiral)
+#'
+#' advs <- dm_peds %>%
+#'   select(USUBJID, BRTHDTC, SEX) %>%
+#'   right_join(., vs_peds, by = "USUBJID") %>%
+#'   filter(USUBJID != "PEDS-1010") %>%
+#'   mutate(
+#'     VSDT = ymd(VSDTC),
+#'     BRTHDT = ymd(BRTHDTC)
+#'   ) %>%
+#'   derive_vars_duration(
+#'     new_var = AGECUR_D,
+#'     new_var_unit = CURU_D,
+#'     start_date = BRTHDT,
+#'     end_date = VSDT,
+#'     out_unit = "days",
+#'     trunc_out = FALSE
+#'   ) %>%
+#'   derive_vars_duration(
+#'     new_var = AGECUR_M,
+#'     new_var_unit = CURU_M,
+#'     start_date = BRTHDT,
+#'     end_date = VSDT,
+#'     out_unit = "months",
+#'     trunc_out = FALSE
+#'   ) %>%
+#'   mutate(
+#'     AGECUR = ifelse(AGECUR_D >= 365.25 * 2, AGECUR_M, AGECUR_D),
+#'     AGECURU = ifelse(AGECUR_D >= 365.25 * 2, CURU_M, CURU_D)
+#'   )
+#'
+#' #' metadata is in months
+#' cdc_meta_criteria <- admiralpeds::cdc_wtage %>%
+#'   mutate(
+#'     age_unit = "months",
+#'     SEX = ifelse(SEX == 1, "M", "F")
+#'   )
+#'
+#' #' metadata is in days
+#' who_meta_criteria <- bind_rows(
+#'   (admiralpeds::who_wt_for_age_boys %>%
+#'     mutate(
+#'       SEX = "M",
+#'       age_unit = "days"
+#'     )
+#'   ),
+#'   (admiralpeds::who_wt_for_age_girls %>%
+#'     mutate(
+#'       SEX = "F",
+#'       age_unit = "days"
+#'     )
+#'   )
+#' ) %>%
+#'   rename(AGE = Day)
+#'
+#' criteria <- bind_rows(
+#'   cdc_meta_criteria,
+#'   who_meta_criteria
+#' ) %>%
+#'   rename(AGEU = age_unit)
+#'
+#' derive_params_growth_age(
+#'   advs,
+#'   sex = SEX,
+#'   age = AGECUR,
+#'   age_unit = AGECURU,
+#'   meta_criteria = criteria,
+#'   parameter = VSTESTCD == "WEIGHT",
+#'   set_values_to_sds = exprs(
+#'     PARAMCD = "WTASDS",
+#'     PARAM = "Weight-for-age z-score"
+#'   ),
+#'   set_values_to_pctl = exprs(
+#'     PARAMCD = "WTAPCTL",
+#'     PARAM = "Weight-for-age percentile"
+#'   )
+#' )
+derive_params_growth_age <- function(dataset,
+                                     sex,
+                                     age,
+                                     age_unit,
+                                     meta_criteria,
+                                     parameter,
+                                     set_values_to_sds = NULL,
+                                     set_values_to_pctl = NULL) {
+  # Apply assertions to each argument to ensure each object is appropriate class
+  # need to add assertion for dataset that checks columns in sex, age, age_unit are present
```

Comment:
having trouble parsing out the correct code here, but its something along the lines of:

`assert_data_frame(dataset, required_vars = expr_c({{sex}}, {{age}}, {{age_unit}})`

Reply:
@bundfussr would you be able to advise Zelos on the code to use here please?

Reply:
It seems to be resolved.

Reply:
By the way, `!!` could be used instead of `{{ }}` when referring to arguments like `sex`, `age`, ... This would be more in line with the other admiral code.

Reply:
thanks Stefan! sorry, missed that this one was an old comment...